src/alias/azext_alias/_const.py

```diff
@@ -11,11 +11,15 @@
 COLLIDED_ALIAS_FILE_NAME = 'collided_alias'
 COLLISION_CHECK_LEVEL_DEPTH = 4
 
-PLACEHOLDER_REGEX = r'\s+{\d+}'
+POS_ARG_NAME_REGEX = r'{{\s*([a-zA-Z0-9_]+)\s*}}'
+EXPRESSION_REGEX = r'{{\s*([a-zA-Z0-9_]+.*?)\s*}}'
+NUMBER_PLACEHOLDER_REGEX = r'{{\s*\d+\s*}}'
```

Comment:
3 times the regex, 3 times the fun.

Reply:
I've removed all the complex regular expressions above and convert them into actual code/function. Though simple regular expressions are still used in some places to simplify implementation.